### ISSUE TITLE
Add support for a iTerm 2-escue 'Smart window placement' option

### ISSUE THREAD

**Issue by cyberpunks-info** [zeit/hyperterm#683]
[iTerm2](https://iterm2.com) has an option called `Smart window placement` (search for it on [this](https://iterm2.com/documentation-one-page.html) page) that places new terminal windows where they will overlap other terminal windows the least. At the moment, Hyperterm place new windows in the center of the screen, which can at times be irritating. An option to place new windows randomly could also be useful.

**Comment by TheSmokingGnu** [comment]
I would like to have a go at this.

I've found the code in the at the iterm2 github here in the smart layout function: https://github.com/gnachman/iTerm2/blob/master/sources/iTermWindowImpl.m

Looks like it just scans the page from the center looking for the least overlap.  A bit expensive I suppose but if this is a config item that is turned off by default it would be a pretty nice feature.

**Comment by ppot** [comment]
@TheSmokingGnu  Go ahead.

**Comment by nicholasc** [comment]
I would take this over since there's no news from @TheSmokingGnu.

I was thinking a simple `smartWindowPlacement` config option that defaults to false would do.

Any additional suggestions?

**Comment by TheSmokingGnu** [comment]
@nicholasc Fair enough, didn't have time to get around to it.  I think all of the helper functions are already made, getting all the current window placements etc.  Not sure what to do about multi-monitor setups though

**Comment by nicholasc** [comment]
@TheSmokingGnu Allright excellent. I could merge from your fork and complete the rest if you're ok with that. I cannot seem to find the branch though.

Also, I am not certain iTerm actually supports multi-monitor smart placement at the moment. From what I see in the code, it just attempts to calculate the less obtrusive position for a new window in the current screen based on other opened terminal windows.

I'll try and think of a way to implement it.

**Comment by ppot** [comment]
@nicholasc There is no branch on this yet.

**Comment by ianmobbs** [comment]
It looks like the code to be changed is:

https://github.com/zeit/hyper/blob/master/app/index.js#L94
